lib/logstash/inputs/udp.rb

```diff
@@ -114,4 +128,21 @@ def stop
     @udp.close rescue nil
   end
 
+  def identity(client)
+    [client[3], client[1]].compact.join("-")
+  end
+
+  def need_identity_map?
+    case @codec
+    when LogStash::Codecs::Netflow
+      true
+    when LogStash::Codecs::IdentityMapCodec
+      true
+    when LogStash::Codecs::Multiline
+      true
+    else
+      false
+    end
+  end
+
```

Comment:
It is non-obvious but the problem with removing LogStash::Codecs::IdentityMapCodec from consideration in `need_identity_map?`is that `need_identity_map?` is also used in the inputworker.
During register `@codec` goes from class Logstash::Codecs::Netflow to LogStash::Codecs::IdentityMapCodec after, so futher down the line in the inputworker `@codec` will actually never be LogStash::Codecs::Netflow, but LogStash::Codecs::IdentityMapCodec.
Since `need_identity_map?` is used in inputworker, it will also need to return true if `@codec` is of class LogStash::Codecs::IdentityMapCodec. Otherwise no identity is passed to the codec, even though it is needed.